### PR TITLE
[ADP-3372] Add getWalletTip endpoint to deposit wallet

### PR DESCRIPTION
**Pull request by paweljakubas**
- [ ] add endpoint skeleton, ToJSON, FromJSON, Schema to `ApiT ChainPoint` and all needed testing

### CODE REVIEW

File: lib/customer-deposit-wallet/http/Cardano/Wallet/Deposit/HTTP/Types/JSON.hs [hunk 1]
```diff
@@ -140,3 +147,27 @@ instance ToSchema (ApiT CustomerList) where
             $ NamedSchema
                 (Just "ApiT CustomerList")
                 customerListSchema
+
+instance ToJSON (ApiT ChainPoint) where
+    toJSON (ApiT Origin) = "origin"
+    toJSON (ApiT (At sl)) = object
+        [ "at_slot" .= toJSON sl
+        ]
```
Comment (HeinrichApfelmus) [inline lib/customer-deposit-wallet/http/Cardano/Wallet/Deposit/HTTP/Types/JSON.hs]:
```suggestion
    toJSON (ApiT Origin) = "genesis"
    toJSON (ApiT (At sl)) = object
        [ "slot_no" .= toJSON sl
        ]
```

The current state of the nomenclature for the number is this:

* `ouroboros-consensus` and `cardano-ledger-core` have settled on `SlotNo`
* The ledger specification calls it `Slot`.

The `ChainPoint` type is transitional architecture, I want to import it from `Cardano.Wallet.Read` instead, specifically here:

https://github.com/cardano-foundation/cardano-wallet-agda/blob/433c08fbdce295cfb8206f47b1a5f19196f6e248/lib/cardano-wallet-read/haskell/Cardano/Wallet/Read/Chain.hs#L55-L61

This suggests the above naming conventions for the fields.

**Review by HeinrichApfelmus — APPROVED** [review]
Excellent, thank you! 😊 I would like to see different naming conventions for the fields, though, based on what we will have in `Cardano.Wallet.Read`. Could you amend that? Thanks!

File: lib/customer-deposit-wallet/http/Cardano/Wallet/Deposit/HTTP/Types/API.hs [hunk 2]
```diff
@@ -37,3 +39,7 @@ type CustomerAPI =
         "customers"
             :> Capture "customerId" (ApiT Customer)
             :> Verb 'PUT 200 '[JSON] (ApiT Address)
+
+type NetworkAPI =
+        "network" :> "tip"
```
Comment (HeinrichApfelmus) [inline lib/customer-deposit-wallet/http/Cardano/Wallet/Deposit/HTTP/Types/API.hs]: Also, we have to be careful about naming here — `getWalletTip` does **not** return the tip of the network, but the point until which the wallet has synchronized. 🤔

File: lib/customer-deposit-wallet/http/Cardano/Wallet/Deposit/HTTP/Types/JSON.hs (hunk 1 above)
Comment (paweljakubas) [reply lib/customer-deposit-wallet/http/Cardano/Wallet/Deposit/HTTP/Types/JSON.hs]: :+1:

File: lib/customer-deposit-wallet/http/Cardano/Wallet/Deposit/HTTP/Types/API.hs (hunk 2 above)
Comment (paweljakubas) [reply lib/customer-deposit-wallet/http/Cardano/Wallet/Deposit/HTTP/Types/API.hs]: indeed, proposed `/network/synchronization-point` which seems to be very clear and correct IMHO

Comment (HeinrichApfelmus) [reply lib/customer-deposit-wallet/http/Cardano/Wallet/Deposit/HTTP/Types/API.hs]: Uh, our typical nomenclature for this is `localTip` of the wallet. 😅 I think that just `/localTip`, `/walletTip` or even `/tip` would be appropriate.

Comment (paweljakubas) [reply lib/customer-deposit-wallet/http/Cardano/Wallet/Deposit/HTTP/Types/API.hs]: `local-tip` is much better. Thanks!

**Pull request merged**